jax/_src/numpy/lax_numpy.py

```diff
@@ -2342,35 +2342,82 @@ def _pad_edge(array, pad_width):
   return array
 
 
-@partial(jit, static_argnums=(1, 2))
-def _pad(array, pad_width, mode, constant_values):
-  array = asarray(array)
+def _pad_stats(array, pad_width, stat_length, stat_func):
   nd = ndim(array)
+  for i in range(nd):
+    if stat_length is None:
+      stat_before = stat_func(array, axis=i, keepdims=True)
+      stat_after = stat_before
+    else:
+      array_length = array.shape[i]
+      length_before, length_after = stat_length[i]
+      if length_before == 0 or length_after == 0:
+        raise ValueError("stat_length of 0 yields no value for padding")
+
+      # Limit stat_length to length of array.
+      if length_before > array_length:
+        length_before = array_length
+      if length_after > array_length:
+        length_after = array_length
```

Comment:
nit: maybe use `length_before = mlin(length_before, array_length)`

Reply:
I tried `length_before = min(length_before, array_length)` and got an error.
I suspect `jax.numpy.min` shadows python function `min`

Reply:
Sorry, yes you need to use `_min()` here to get the builtin `min()`.

Reply:
Oh I see! updated ;)